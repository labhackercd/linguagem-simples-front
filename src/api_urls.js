module.exports = Object.freeze({

    //NAME CONSTANTS
    APPLICATION_NAME: "Estudio",

    //APPLICATION URL'S
        APPLICATION_FRONT_BASE_URL: "www.teste.com",  //APPLICATION FRONT BASE URL
        //APPLICATION_SERVER_API_BASE_URL: 'http://localhost:8000/api/', //APPLICATION API SERVER BASE URL
        APPLICATION_SERVER_API_BASE_URL: 'https://acompanhe.camara.leg.br/server/api/',
        APPLICATION_RESET_PASSWORD_URL: 'https://acompanhe.camara.leg.br/server/password_reset/',

    //ROUTES
        INITIAL_PAGE_URL: '/', //Application Login and Initial page
        ESTUDIO_PAGE_URL: '/estudio', //Application Estudio Page
        DASHBOARD_PAGE_URL: '/dashboard/:dashboardId', //Application Dashboard page
        DASHBOARD_BASE_URL: '/dashboard/',

    //API'S CAMARA DOS DEPUTADOS
        API_CD_EVENTOS_PLENARIO: 'https://dadosabertos.camara.leg.br/api/v2/eventos?codTipoEvento=110',// 110 code corresponds to Plenary
        API_CD_EVENTOS: 'https://dadosabertos.camara.leg.br/api/v2/eventos/',
        API_CD_ORIENTACAO_VOTACAO: 'https://liderancadigital.camara.leg.br/proxy_pass/secod-wsVotDecom-orientacao',
        API_CD_ITENS_EM_VOTACAO: 'https://infoleg.camara.gov.br/wsVotDecom/votacao/itens-em-votacao-na-reuniao/',

    //API'S BACKEND
        //AUTH URL'S
            TOKEN_REFRESH_URL: '/token/refresh/', //Application Api Token REFRESH PATH
            TOKEN_OBTAIN_URL: '/token/obtain/', //Application Api Token PATH
            TOKEN_VERIFY_URL: '/token/verify/', //Application Api Token Verify

        //FILE-VIDEO
            API_FILE_VIDEO_URL:'/file-video/', //Application end-point to obtain the source of specific video

        //NEWS
            API_NEWS_URL:'/newsâ€‹/news_list', //Obtain news list
            API_NEWS_SEARCH_URL:'/news-search/', //Obtain specific news using parameters

        //PUBLICATIONS
            API_PUBLICATIONS_URL:'/publications/',

        //RADIO AGENCY
            API_RADIO_AGENCY_URL:'/radioagency/',
            API_RADIO_AGENCY_SEARCH_URL:'/radioagency-search/',

        //RADIO CAMARA
            API_RADIO_CAMARA_URL:'/radiocamara/',
            API_RADIO_CAMARA_SEARCH_URL:'/radiocamara-search/',

        //SAVED CONTENTS
            API_SAVED_CONTENTS_URL:'/saved-contents/',

        //SESSIONS
            API_SESSIONS_URL:'/sessions/',

        //TV CAMARA
            API_TV_CAMARA_URL:'/tvcamara/',
            API_TV_CAMARA_SEARCH_URL:'/tvcamara-search/',

        //VIDEOS
            API_VIDEOS_SNIPPETS:'/videos-session/'
  });
