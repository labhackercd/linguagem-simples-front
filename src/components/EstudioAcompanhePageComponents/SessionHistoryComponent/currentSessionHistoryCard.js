import React from "react";
import Box from "@material-ui/core/Box";
import Grid  from "@material-ui/core/Grid";
import Typography  from "@material-ui/core/Typography";
import Paper from '@material-ui/core/Paper'
import { makeStyles } from '@material-ui/core/styles';
import StartedIcon from './assets/session_started.svg'
import 'date-fns';
import FormatStringData from './utils';
import FormatListBulletedIcon from '@material-ui/icons/FormatListBulleted';


const useStyles = makeStyles({
  root: {
    background:'#007E5A',
    border: 0,
    height: '100%',
    width: '100%',
    position: "fixed",
    display: "flex"
  },
  sessionList:{
    overflow: 'auto',
    maxHeight: 350,
    maxWidth: '100%'
  },
  input:{
      color: "green"
  },
  sessionHistoryCard:{
    background:'#F4F4F4',
  },
  currentSessionHistoryCard:{
    background:'#00AF82',
    color: 'white',
  },
  iconCurrentSessionHistoryCard:{
    background:'#007E5A',
    margin:5
  },
  iconSessionHistoryCard:{
    background:'#FFFFFF',
    margin:5
  },
  boxPaper:{
      padding:"5px"
  }
});



export default function CurrentSessionHistoryCard(props){
    const classes = useStyles();
    const sessionDate = FormatStringData(props.infoSession.date);

    return (
        <Paper elevation={0} className={classes.currentSessionHistoryCard}>
            <Grid container>
                <Grid item xs={2}>
                    <Paper elevation={0} className={classes.iconCurrentSessionHistoryCard}>
                        <Box display="flex" alignItems="center" justifyContent="center" className={classes.boxPaper}>
                            <FormatListBulletedIcon fontSize="large" style={{ color: "#FFFFFF" }}></FormatListBulletedIcon>
                        </Box>
                    </Paper>
                </Grid>
                <Grid item xs={10}>
                    <Box m={1}>
                        <Grid container>
                            <Grid item xs={6}>
                                <Typography variant="body1">{sessionDate}</Typography>
                            </Grid>
                            <Grid item xs={6}>
                                <Box display="flex" justifyContent="flex-end">
                                    <img src={StartedIcon} alt="Status sessão iniciada"></img>
                                </Box>
                            </Grid>
                            <Grid item xs={12}>
                                <Typography variant="h6">Sessão Deliberativa Extraordinária</Typography>
                            </Grid>
                        </Grid>
                    </Box>
                </Grid>
            </Grid>
        </Paper>
    );
}