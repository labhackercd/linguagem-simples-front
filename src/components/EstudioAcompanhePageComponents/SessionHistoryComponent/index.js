import React, { useState, useEffect } from "react";
import Box from "@material-ui/core/Box";
import Grid  from "@material-ui/core/Grid";
import Typography  from "@material-ui/core/Typography";
import Paper from '@material-ui/core/Paper'
import { makeStyles } from '@material-ui/core/styles';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import SvgIcon from '@material-ui/core/SvgIcon';
import FormatListBulletedIcon from '@material-ui/icons/FormatListBulleted';
import { ReactComponent as DocumentIcon } from './assets/document_icon.svg';
import ScheduleIcon from './assets/schedule_session.svg'
import FinishedIcon from './assets/finished_session.svg'
import StartedIcon from './assets/session_started.svg'
import SessionHappening from './assets/session_happening_icon.svg'
import CircularProgress from '@material-ui/core/CircularProgress';

import 'date-fns';
import DateFnsUtils from '@date-io/date-fns';
import { MuiPickersUtilsProvider, KeyboardDatePicker,
} from '@material-ui/pickers';
import ptBrLocale from "date-fns/locale/pt-BR";

import axiosInstance from '../../../auth/axiosApi.js'


const useStyles = makeStyles({
  root: {
    background:'#007E5A',
    border: 0,
    height: '100%',
    width: '100%',
    position: "fixed",
    display: "flex"
  },
  sessionList:{
    overflow: 'auto',
    maxHeight: 350,
    maxWidth: '100%'
  },
  input:{
      color: "green"
  },
  sessionHistoryCard:{
    background:'#F4F4F4',
  },
  currentSessionHistoryCard:{
    background:'#00AF82',
    color: 'white',
  },
  iconCurrentSessionHistoryCard:{
    background:'#007E5A',
    margin:5
  },
  iconSessionHistoryCard:{
    background:'#FFFFFF',
    margin:5
  },
  boxPaper:{
      padding:"5px"
  }
});


function ScheduleOrFinishedSessionHistoryCard(props){
    const classes = useStyles();
    const sessionStatus = props.status;

    return (
        <Paper elevation={0} className={classes.sessionHistoryCard}>
            <Grid container>
                <Grid item xs={2}>
                    <Paper elevation={0} className={classes.iconSessionHistoryCard}>
                        <Box display="flex" alignItems="center" justifyContent="center" className={classes.boxPaper}>                           
                            {sessionStatus === "pre_session" ? 
                                <AccessTimeIcon fontSize="large" style={{ color: "FAC915" }}></AccessTimeIcon>
                                : 
                                <SvgIcon fontSize="large"><DocumentIcon></DocumentIcon></SvgIcon>
                            }
                        </Box>
                    </Paper>
                </Grid>
                <Grid item xs={10}>
                    <Box m={1}>
                        <Grid container>
                            <Grid item xs={6}>
                                <Typography variant="body1">17/07</Typography>
                            </Grid>
                            <Grid item xs={6}>
                                <Box display="flex" justifyContent="flex-end">
                                {sessionStatus === "pre_session" ? 
                                    <img src={ScheduleIcon} alt="Status sessão agendada"></img>
                                    : 
                                    <img src={FinishedIcon} alt="Status sessão encerrada"></img>
                                }
                                    
                                </Box>
                            </Grid>
                            <Grid item xs={12}>
                                <Typography variant="h6" style={{ color: "grey" }}>Sessão Deliberativa Extraordinária</Typography>
                            </Grid>
                        </Grid>
                    </Box>
                </Grid>
            </Grid>
        </Paper>
    );
}

function CurrentSessionHistoryCard(props){
    const classes = useStyles();

    return (
        <Paper elevation={0} className={classes.currentSessionHistoryCard}>
            <Grid container>
                <Grid item xs={2}>
                    <Paper elevation={0} className={classes.iconCurrentSessionHistoryCard}>
                        <Box display="flex" alignItems="center" justifyContent="center" className={classes.boxPaper}>                           
                            <FormatListBulletedIcon fontSize="large" style={{ color: "#FFFFFF" }}></FormatListBulletedIcon>
                        </Box>
                    </Paper>
                </Grid>
                <Grid item xs={10}>
                    <Box m={1}>
                        <Grid container>
                            <Grid item xs={6}>
                                <Typography variant="body1">17/07</Typography>
                            </Grid>
                            <Grid item xs={6}>
                                <Box display="flex" justifyContent="flex-end">
                                    <img src={StartedIcon} alt="Status sessão agendada"></img>
                                </Box>
                            </Grid>
                            <Grid item xs={12}>
                                <Typography variant="h6">Sessão Deliberativa Extraordinária</Typography>
                            </Grid>
                        </Grid>
                    </Box>
                </Grid>
            </Grid>
        </Paper>
    );
}


class SessionHistoryComponent extends React.Component {
    constructor(props){
        super(props);
        this.state = {
            sessionsList: [],
            dataLoaded:false,
        };

    }

    loadSessions(callback){
        axiosInstance.get('/sessions/', {
        }).then(
            result => {
                console.log(result)
                console.log(result.data)
                if(result.status===200){
                    this.setState({sessionsList:result.data})
                }else{
                    alert("Erro ao recuperar sessões");
                }
                callback()
            } 
        ).catch (error => {
            throw error;
        })
    }

    componentDidMount(){
        this._isMounted = true;

        if(this._isMounted){
            this.loadSessions(() => {
                console.log("Data:" + this.state.sessionsList)
                this.setState({dataLoaded:true});
 
            });
        }
    }


    render(){
        const { classes } = this.props;

        if(!this.state.dataLoaded){
            return (<Box display="flex" justifyContent="center" alignItems="center">
                        <CircularProgress></CircularProgress>
                    </Box>)
        }

        return(
            <Box>
                <Grid container>
                    <Grid item xs={12}>
                            <Grid container>
                                <Grid item xs={6}>
                                    <Box display="flex" justifyContent="flex-start" paddingBottom={3}>
                                        <Typography variant="h4" color="textSecondary">Histórico</Typography>
                                    </Box>
                                </Grid>
                                <Grid item xs={6}>
                                    <Box display="flex" justifyContent="flex-end" paddingBottom={3}>
                                        <MuiPickersUtilsProvider utils={DateFnsUtils} locale={ptBrLocale}>
                                        <KeyboardDatePicker
                                            disableToolbar
                                            variant="inline"
                                            value=""
                                            format="dd/MM/yyyy"
                                            id="date-picker-search"
                                            InputLabelProps={{
                                                shrink: true,
                                            }}
                
                                            />
                                        </MuiPickersUtilsProvider>
                                    </Box>
                                </Grid>
                            </Grid>
                    </Grid>
                    <Grid item xs={12}>
                        <Box paddingLeft={2}>
                        <img src={SessionHappening} alt="Uma sessão em andamento"></img>
                        </Box>

                        <List className={classes.sessionList} disablePadding dense={true}>
                        {this.state.sessionsList.map((session) =>
                            <ListItem key={session.id} value={session}> 
                                { (session.situation_session === "pre_session" || session.situation_session === "closed_session") ?
                                    <ScheduleOrFinishedSessionHistoryCard status={session.situation_session}></ScheduleOrFinishedSessionHistoryCard> :
                                    <CurrentSessionHistoryCard></CurrentSessionHistoryCard>
                                }
                            </ListItem>
                        )}

                        </List>
                    </Grid>
                </Grid>
                <Box pt={2}>
                    
                </Box>
            </Box>
        )
    }
}

export default withStyles(useStyles)(SessionHistoryComponent);