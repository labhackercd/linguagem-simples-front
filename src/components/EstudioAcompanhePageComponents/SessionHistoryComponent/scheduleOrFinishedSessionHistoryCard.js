import React from "react";
import Box from "@material-ui/core/Box";
import Grid  from "@material-ui/core/Grid";
import Typography  from "@material-ui/core/Typography";
import Paper from '@material-ui/core/Paper'
import { makeStyles } from '@material-ui/core/styles';
import AccessTimeIcon from '@material-ui/icons/AccessTime';
import SvgIcon from '@material-ui/core/SvgIcon';
import { ReactComponent as DocumentIcon } from './assets/document_icon.svg';
import ScheduleIcon from './assets/schedule_session.svg'
import FinishedIcon from './assets/finished_session.svg'
import 'date-fns';
import FormatStringData from './utils'

const useStyles = makeStyles({
    sessionHistoryCard:{
      background:'#F4F4F4',
    },
    iconCurrentSessionHistoryCard:{
      background:'#007E5A',
      margin:5
    },
    iconSessionHistoryCard:{
      background:'#FFFFFF',
      margin:5
    },
    boxPaper:{
        padding:"5px"
    }
  });
  

export default function ScheduleOrFinishedSessionHistoryCard(props){
    const classes = useStyles();
    const sessionStatus = props.infoSession.situation_session;
    const sessionDate = FormatStringData(props.infoSession.date);


    return (
        <Paper elevation={0} className={classes.sessionHistoryCard}>
            <Grid container>
                <Grid item xs={2}>
                    <Paper elevation={0} className={classes.iconSessionHistoryCard}>
                        <Box display="flex" alignItems="center" justifyContent="center" className={classes.boxPaper}>
                            {sessionStatus === "pre_session" ?
                                <AccessTimeIcon fontSize="large" style={{ color: "FAC915" }}></AccessTimeIcon>
                                :
                                <SvgIcon fontSize="large"><DocumentIcon></DocumentIcon></SvgIcon>
                            }
                        </Box>
                    </Paper>
                </Grid>
                <Grid item xs={10}>
                    <Box m={1}>
                        <Grid container>
                            <Grid item xs={6}>
                                <Typography variant="body1">{sessionDate}</Typography>
                            </Grid>
                            <Grid item xs={6}>
                                <Box display="flex" justifyContent="flex-end">
                                {sessionStatus === "pre_session" ?
                                    <img src={ScheduleIcon} alt="Status sessão agendada"></img>
                                    :
                                    <img src={FinishedIcon} alt="Status sessão encerrada"></img>
                                }
                                </Box>
                            </Grid>
                            <Grid item xs={12}>
                                <Typography variant="h6" style={{ color: "grey" }}>Sessão Deliberativa Extraordinária</Typography>
                            </Grid>
                        </Grid>
                    </Box>
                </Grid>
            </Grid>
        </Paper>
    );
}
