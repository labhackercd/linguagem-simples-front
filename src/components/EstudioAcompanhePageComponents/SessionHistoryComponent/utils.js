
export default function FormatStringData(date) {

    if(date === null || date === ""){
        return "";
    }
    var day  = date.split("-")[2];
    var month  = date.split("-")[1];
    //var year  = data.split("-")[0];
  
    return ("0"+day).slice(-2) +  '/' + ("0"+month).slice(-2) ;
    // Utilizo o .slice(-2) para garantir o formato com 2 digitos.
}