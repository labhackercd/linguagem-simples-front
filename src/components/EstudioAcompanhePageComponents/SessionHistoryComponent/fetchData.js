import axiosInstance from '../../../auth/axiosApi.js'
import {API_SESSIONS_URL_NOT_LIVE, API_SESSIONS_CURRENT_LIVE_URL} from '../../../api_urls'

export async function fetchData() {

  const response = await axiosInstance.get(API_SESSIONS_URL_NOT_LIVE);
  
  //console.log("CODE: " +response.status)
  
  if(response.status===200){
    return response.data;
  }else{
    return null;
  }

};


export async function fetchCurrentSessionsData() {

  const response = await axiosInstance.get(API_SESSIONS_CURRENT_LIVE_URL);
  //console.log(response)
  
  if(response.data.length > 0){
    return response.data;
  }else{
    return null;
  }

};