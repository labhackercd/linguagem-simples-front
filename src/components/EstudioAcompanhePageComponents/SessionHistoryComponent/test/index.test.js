import React from 'react';
import {shallow} from "enzyme/build";
import { mount} from 'enzyme';
import MockAdapter from "axios-mock-adapter"
import axiosInstance from '../../../../auth/axiosApi'

import CircularProgress from '@material-ui/core/CircularProgress';
import SessionHistoryComponent from '../index.js'
import {API_SESSIONS_URL_NOT_LIVE, API_SESSIONS_CURRENT_LIVE_URL} from '../../../../api_urls'


it("should render the SessionHistoryComponent component and match snapshot", () => {
    const component = mount(<SessionHistoryComponent/>);

    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});


describe('Testing lifeclycle of SessionHistoryComponent content', () => {
   
  test("Test SessionHistoryComponent content lifeclycle when data is not already loaded", async (done) => {
                 
      const wrapper = await mount(<SessionHistoryComponent/>);

      setImmediate(() => {
        wrapper.update();
        //console.log(wrapper.debug())
        const containsSpinner = wrapper.containsMatchingElement(<CircularProgress />);
        expect(containsSpinner).toBeTruthy()
        done();
     })

  });
 
  test("Test SessionHistoryComponent content lifeclycle with full data been loaded", async (done) => {
    var mockInstance = new MockAdapter(axiosInstance);
    const currentLiveSessions = [
        {
          "id": 1,
          "author": {
            "id": 1,
            "is_superuser": true,
            "username": "admin",
            "first_name": "",
            "last_name": "",
            "email": "a@a.com",
            "profile": "editor"
          },
          "location": "plenary",
          "date": "2020-09-24",
          "type_session": "virtual",
          "situation_session": "pre_session",
          "resume": "Resumo",
          "enable": true,
          "id_session_dados_abertos": null
        }
      ];
    const currentNotLiveSessions = [
        {
          "id": 2,
          "author": {
            "id": 1,
            "is_superuser": true,
            "username": "admin",
            "first_name": "",
            "last_name": "",
            "email": "a@a.com",
            "profile": "editor"
          },
          "location": "plenary",
          "date": "2020-09-24",
          "type_session": "virtual",
          "situation_session": "pre_session",
          "resume": "",
          "enable": false,
          "id_session_dados_abertos": null
        }
    ];

    await mockInstance.onGet(API_SESSIONS_URL_NOT_LIVE).reply(200,currentNotLiveSessions).onGet(API_SESSIONS_CURRENT_LIVE_URL).reply(200,currentLiveSessions)
    const wrapper = await mount(<SessionHistoryComponent/>);

    setImmediate(() => {
      wrapper.update();
      //console.log(wrapper.debug())
      const containsSpinner = wrapper.containsMatchingElement(<CircularProgress />);
      expect(containsSpinner).not.toBeTruthy()

      //console.log(wrapper.debug())
      done();
   })

  });

  test("Test SessionHistoryComponent content lifeclycle error", async (done) => {
    var mockInstance = new MockAdapter(axiosInstance);

    await mockInstance.onGet(API_SESSIONS_URL_NOT_LIVE).reply(400)
    const wrapper = await mount(<SessionHistoryComponent/>);

    setImmediate(() => {
        wrapper.update();
        done();
     })
   
  });
  
});