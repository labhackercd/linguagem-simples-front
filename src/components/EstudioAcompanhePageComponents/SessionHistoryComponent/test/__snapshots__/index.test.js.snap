// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the SessionHistoryComponent component and match snapshot 1`] = `
<WithStyles(SessionHistoryComponent)>
  <SessionHistoryComponent
    classes={
      Object {
        "sessionList": "SessionHistoryComponent-sessionList-2",
      }
    }
  >
    <Styled(MuiBox)
      alignItems="center"
      display="flex"
      justifyContent="center"
    >
      <div
        className="MuiBox-root MuiBox-root-3"
      >
        <WithStyles(ForwardRef(CircularProgress))>
          <ForwardRef(CircularProgress)
            classes={
              Object {
                "circle": "MuiCircularProgress-circle",
                "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                "circleStatic": "MuiCircularProgress-circleStatic",
                "colorPrimary": "MuiCircularProgress-colorPrimary",
                "colorSecondary": "MuiCircularProgress-colorSecondary",
                "indeterminate": "MuiCircularProgress-indeterminate",
                "root": "MuiCircularProgress-root",
                "static": "MuiCircularProgress-static",
                "svg": "MuiCircularProgress-svg",
              }
            }
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </ForwardRef(CircularProgress)>
        </WithStyles(ForwardRef(CircularProgress))>
      </div>
    </Styled(MuiBox)>
  </SessionHistoryComponent>
</WithStyles(SessionHistoryComponent)>
`;
